src/ledger/autofill.ts

```diff
@@ -25,11 +25,13 @@ interface ClassicAccountAndTag {
  * @param signersCount - The expected number of signers for this transaction. Used for multisign.
  * @returns An autofilled transaction.
  */
-async function autofill(
+async function autofill<T extends Transaction>(
```

Comment:
If you want it to extend, it might need to be `<T extends BaseTransaction>` so that it can handle other transaction types

Reply:
There shouldn't be any "other transaction types", though, right? Or are you thinking about someone using xrpl.js on a sidechain or something?

Reply:
Yeah something like that - by `T extends Transaction` did you mean if someone added extra fields or something?

Reply:
I wrote `T extends Transaction` so that `autofill(Payment)` returns a `Payment` type and not a `Transaction` type.

Reply:
Other than that, theres no difference between this and the original type signature.